src/main/scala/nl/knaw/dans/easy/fsrdb/FsRdbUpdater.scala

```diff
@@ -138,15 +170,18 @@ object FsRdbUpdater {
       .filter(pid => namespaces.exists(pid.startsWith))
   }
 
-  private def updateDB(items: List[Item])(implicit s: Settings): Try[Unit] = Try {
-    val conn = DriverManager.getConnection(s.postgresURL)
+  private def updateDB(conn: Connection, items: List[Item])(implicit s: Settings): Try[Unit] = Try {
     try {
       items.foreach {
         case folder: FolderItem => updateOrInsertFolder(conn, folder).get
         case file: FileItem => updateOrInsertFile(conn, file).get
       }
-    } finally {
-      conn.close()
+      conn.commit(); // end transaction
```

Comment:
Superfluous comments. The terms `commit` and `rollback` should be sufficiently familiar to almost all programmers.